### PR TITLE
Replace usages of `Age` in `EditPatientEffectHandler` with `PatientAgeDetails`

### PR DESCRIPTION
**Pull request by vinaysshenoy**
https://app.clubhouse.io/simpledotorg/story/4484/remove-the-age-class

### CODE REVIEW

File: app/src/main/java/org/simple/clinic/editpatient/EditPatientEffectHandler.kt [hunk 1]
```diff
@@ -203,7 +203,7 @@ class EditPatientEffectHandler @AssistedInject constructor(
   private fun coerceAgeFrom(recordedAgeDetails: PatientAgeDetails, enteredAge: String): PatientAgeDetails {
     val enteredAgeValue = enteredAge.toInt()
     return when {
-      recordedAgeDetails.type == FROM_AGE && recordedAgeDetails.ageValue == enteredAgeValue -> recordedAgeDetails
```
Comment (Honey14) [inline app/src/main/java/org/simple/clinic/editpatient/EditPatientEffectHandler.kt]: We will be removing the types usage entirely, right?

Comment (vinaysshenoy) [reply app/src/main/java/org/simple/clinic/editpatient/EditPatientEffectHandler.kt]: Yeah, that is the eventual plan.

File: app/src/main/java/org/simple/clinic/patient/PatientAgeDetails.kt [hunk 2]
```diff
@@ -36,6 +36,9 @@ data class PatientAgeDetails(
       else -> throw IllegalStateException("Could not infer type from [Age: $ageValue, Age updated: $ageUpdatedAt, DOB: $dateOfBirth]")
     }
 
+  val isRecordedAsAge: Boolean
```
[the remark this answers is not in the document]
Comment (vinaysshenoy) [reply app/src/main/java/org/simple/clinic/patient/PatientAgeDetails.kt]: Good catch. Done.

**Review by Honey14 — APPROVED** [review]
Looks good

**Pull request merged**